File: 백준/Silver/5567. 결혼식/결혼식.cc
#include <iostream>
#include <vector>
#include <queue>

using namespace std;

int ret;
int N,M;
vector<int> v[504];
int visited[504];

void BFS_iter(int start) {
    queue<int> q;
    q.push(start);
    visited[start]=1;
    
    while(!q.empty()) {
        start=q.front();
        q.pop();
        
        for(int temp:v[start]) {
            if(visited[temp]==0) {
                visited[temp]=visited[start]+1;
                q.push(temp);
                
                if(visited[temp]==2 or visited[temp]==3) {
                    ret+=1;
                }
            }
        }
    }
}
int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);
    
    cin >> N;
    cin >> M;
    
    for(int i=0;i<M;i++) {
        int start,end;
        cin >> start >> end;
        v[start].push_back(end);
        v[end].push_back(start);
    }
    
    BFS_iter(1);
    cout << ret << "\n";
    
    return 0;
}

File: 백준/Silver/5567. 결혼식/결혼식.java


import java.util.LinkedList;
import java.util.List;
import java.util.Queue;
import java.util.Scanner;

public class Main {

    static int N,M;

    static int ret;
    static List<Integer> v[];
    static int visited[]=new int[504];

    public static void init_() {

        v=new LinkedList[N+1];
        for(int i=1;i<=N;i++) {
            v[i]=new LinkedList<>();
        }
    }

    public static void BFS_iter(int start) {
        Queue<Integer> q=new LinkedList<>();
        q.add(start);
        visited[start]=1;

        while(!q.isEmpty()) {
            start=q.poll();

            for(int temp:v[start]) {
                if(visited[temp]==0) {
                    visited[temp]=visited[start]+1;

                    if(visited[temp]==2 || visited[temp]==3) {
                        ret += 1;
                    }
                    q.add(temp);
                }
            }
        }
    }

    public static void main(String[] args) {

        Scanner in = new Scanner(System.in);
        N = in.nextInt();
        M = in.nextInt();

        init_();

        for (int i = 0; i < M; i++) {
            int start, end;
            start = in.nextInt();
            end = in.nextInt();
            v[start].add(end);
            v[end].add(start);
        }

        BFS_iter(1);

        System.out.println(ret);
    }
}
